server/controllers/middlewares.js

```diff
@@ -152,6 +153,9 @@ export const apiKey = (req, res, next) => {
 
 /**
  * Authenticate.
+ * @PRE: Need to submit a login (username or email)/password as a POST request.
+ *       api_key is also required
+ * @POST: req.remoteUser and req.user are set to the logged in user
```

Comment:
Yes we should only have `req.remoteUser` but I didn't want to open that can of worms now.